doc/source/learning/kubernetes.rst

```diff
@@ -0,0 +1,213 @@
+Kubernetes
+=====================
+
+List of interesting links:
+---------------------
+
+* Kubernetes official webpage:
+    https://kubernetes.io
+* Kubernetes documentation:
+    https://kubernetes.io/docs/home/
+* Kubernetes basic tutorial:
+    https://kubernetes.io/docs/tutorials/kubernetes-basics/
+* Minikube git page:
+    https://github.com/kubernetes/minikube
+* Minikube tutorial:
+    https://kubernetes.io/docs/setup/learning-environment/minikube/
+* Kubectl cheatsheet:
+    https://kubernetes.io/docs/reference/kubectl/cheatsheet/
+
+An easy way to understand kubernetes is following the tutorial "kubernetes-basics"
+https://kubernetes.io/docs/tutorials/kubernetes-basics/ that provides an interactive console to
+tests the commands of the tutorial. However it is also interesting to have installed a minikube in
+your local machine or in a virtual machine to increase the experience
+
+Keys to learn
+---------------------
+
+ * Deployment
+ * Pod
+ * Service
+ * labels
+ * Scale your app
+ * Rolling updates
+
+
+Deployment
+++++++++++++++++
+
+Create a deployment
+
+.. code-block:: console
+
+  kubectl run <myDeployment> --image=<myImage>:<version> --port=8080
+
+Check current deployments
+
+.. code-block:: console
+
+  kubectl get deployments
+
+Create a proxy. The ports will be accessibles from outside the pod
+
+.. code-block:: console
+
+  kubectl proxy
+
+
+Access to the pod proxy
+
+.. code-block:: console
+
+  curl http://localhost:8001/api/v1/namespaces/default/pods/<pod name>/proxy/
+
+
+Pods
+++++++++++++++++
+
+Get pods information
+
+.. code-block:: console
+
+  kubectl get pods
+  kubectl describe pods
+  kubectl get pods -o wide
+
+Get pod log
+
+.. code-block:: console
+
+  kubectl logs <pod name>
+
+Execute command inside the pod
+
+.. code-block:: console
+
+  kubectl exec <pod name> <command>
+
+NOTE: The name of the container can be omitted if there is only one container.
+
+Open a bash session
+
+.. code-block:: console
+
+  kubectl exec -ti <pod name> bash
+
+Services
+++++++++++++++++
+
+Get services
+
+.. code-block:: console
+
+  kubectl get services
+  kubectl describe services
+
+Create a new service (expose it as nodeport)
```

Comment:
I would point to the different types of services here. It's a big deal to understand how networking works.

Reply:
I added a link to https://kubernetes.io/docs/concepts/services-networking/service/ that I think is very clear explaining these concepts

Reply:
The exact page I thought about.